client/src/routes/Quiz.tsx

```diff
@@ -0,0 +1,95 @@
+import { Link } from "react-router-dom";
+import { useLoaderData } from "react-router-typesafe";
+import { getRelativeTime } from "../utils";
+import { quizLoader } from "./Quiz.data";
+import { useState } from "react";
+import { is } from "valibot";
+
+export const Quiz = () => {
+    const [currentQuestion, setCurrentQuestion] = useState(0);
+    const [selectedOption, setSelectedOption] = useState("");
+    const quiz = useLoaderData<typeof quizLoader>();
+
+    const handleOptionChange = (optionId: string) => {
+        setSelectedOption(optionId);
+        const updatedQuiz = { ...quiz };
+        updatedQuiz.questions[currentQuestion].selected_option_id = optionId;
+
+        // If you need to do something with updatedQuiz, like saving it, you can handle that here
+    };
+    const isPrevDisabled = () => {
+        return currentQuestion === 0;
+    };
+    const isNextDisabled = () => {
+        return (
+            currentQuestion === quiz.questions.length - 1 &&
+            selectedOption === ""
```

Comment:
why do we have these as functions instead of variables?
just assign it to what's returned.

`const isPrevDisabled = currentQuestion === 0`